packages/rsbuild-plugin-angular/src/lib/models/normalize-options.unit.test.ts

```diff
@@ -0,0 +1,192 @@
+import { describe, expect } from 'vitest';
+import {
+  getHasServer,
+  normalizeOptions,
+  resolveFileReplacements,
+} from './normalize-options.ts';
+import { vol } from 'memfs';
+
+import { MEMFS_VOLUME } from '@ng-rspack/testing-utils';
+import path from 'node:path';
+
+describe('resolveFileReplacements', () => {
+  it('should work with empty results', () => {
+    const root = '/root';
+
+    expect(resolveFileReplacements([], root)).toStrictEqual([]);
+  });
+
+  it('should return the given FileReplacement with `replace` and `with` paths resolved against the given root', () => {
```

Comment:
We could improve this example by showing that resolve logic is applied, not only that the path starts from root.